pymbolic/primitives.py

```diff
@@ -686,88 +739,64 @@ class Leaf(AlgebraicLeaf):
     pass
 
 
+@augment_attrs_expression_class
+@attrs.frozen(cache_hash=True, init=True)
 class Variable(Leaf):
     """
     .. attribute:: name
     """
-    init_arg_names = ("name",)
-
-    def __init__(self, name):
-        assert name
-        self.name = intern(name)
```

Comment:
We're no longer auto-interning. I'm not sure of the performance impact. Likely minor. See `make_variable` below.

Reply:
Let's assume that's OK unless we learn otherwise.